utils/generate_grammar_tables.py

```diff
@@ -339,20 +348,26 @@ def main():
         assert len(cap_dict) == 1
         populate_capability_bit_mapping_dict(cap_dict[0])
 
-        print(generate_instruction_table(grammar['instructions'], False),
-              file=open(args.core_insts_output, 'w'))
-        print(generate_operand_kind_table(grammar['operand_kinds']),
-              file=open(args.operand_kinds_output, 'w'))
-
-    with open(args.extinst_glsl_grammar) as json_file:
-        grammar = json.loads(json_file.read())
-        print(generate_instruction_table(grammar['instructions'], True),
-              file=open(args.glsl_insts_output, 'w'))
-
-    with open(args.extinst_opencl_grammar) as json_file:
-        grammar = json.loads(json_file.read())
-        print(generate_instruction_table(grammar['instructions'], True),
-              file=open(args.opencl_insts_output, 'w'))
+        if args.core_insts_output is not None:
+            print(generate_instruction_table(grammar['instructions'], False),
+                  file=open(args.core_insts_output, 'w'))
+        if args.operand_kinds_output is not None:
+            print(generate_operand_kind_table(grammar['operand_kinds']),
+                  file=open(args.operand_kinds_output, 'w'))
+
+    if args.extinst_glsl_grammar is not None:
+        with open(args.extinst_glsl_grammar) as json_file:
+            grammar = json.loads(json_file.read())
+            if args.glsl_insts_output is not None:
```

Comment:
Would it make more sense to check this condition right on line 358?  What's the point of loading a grammar file if no output will be produced from it?